Newsletters: add 405 (2026-05-15) (#2752)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2026-05-15-newsletter.md

- [BIPs #2116][] publishes [BIP323][], which proposes expanding the number of
  bits available in `nVersion`'s nonce space for miners from 16 to 24,
  superseding [BIP320][]. It reserves bits 5 through 28 for header-only mining
  without relying on rolling `nTime` more often than once per second. See
  [Newsletter #395][news395 nversion] for previous discussion.

- [BDK #2195][] fixes syncing from Electrum servers when a transaction's first
  output isn't indexed, such as an `OP_RETURN` output. Previously,
  `BdkElectrumClient::populate_with_txids` queried confirmation history using
  the first output's script, which could return an empty history. BDK now uses
  the first indexed output script, or falls back to an input's previous output
  script if none of the outputs are indexed.

- [Bitcoin Inquisition #100][] implements [BIP446][]'s `OP_TEMPLATEHASH` opcode
  for testing proposed consensus changes on [signet][topic signet].
  `OP_TEMPLATEHASH` is a [tapscript][topic tapscript] opcode that pushes a hash
  of the spending transaction onto the stack (see
  [Newsletter #397][news397 templatehash]). The PR also adds an extensive test
  framework.

> - [BINANA #20][] assigns BIN-2026-0002 to a future Bitcoin Inquisition

Reviewer: ```suggestion
- [BINANAs #20][] assigns BIN-2026-0002 to a future Bitcoin Inquisition
```

Broken link

  The vulnerability was discovered and responsibly disclosed by Cory Fields, who
  also provided a proof of concept and proposed mitigation. The issue was fixed
  in Bitcoin Core 29.0.

- **BIP proposal for UTXO set sharing over P2P network**: Fabian Jahr
  [posted][p2p share ml] to the Bitcoin-Dev mailing list about his
  [draft BIP][BIPs #2137] for sharing the UTXO set over the P2P layer.
  The goal of the proposal is to improve the [assumeUTXO][topic assumeutxo]
  feature by providing a way for new nodes to receive the UTXO set directly
  from peers, instead of from external sources. In particular, the proposal
  defines an extension to the P2P protocol which introduces a new service bit,
  four new P2P messages, and merkle root known to the requesting node, to
  verify the correctness of the provided UTXO set.

  The proposal received some interesting feedback. Antoine Riard proposed to
  build the current draft of top of [BIP434][], which defines peer feature negotiation
  (see [Newsletter #386][news386 feat negot]), and brought up some concerns about
  malicious peers forwarding a malformed UTXO set. Eric Voskuil warned the author on

>   the long-term risks of such a BIP, which coul lead to new proposals for miner

Reviewer 2: ```suggestion
  the long-term risks of such a BIP, which could lead to new proposals for miner
```

  through nodes that advertise onion message support using feature bits 38/39.
  Since onion messages are not payments, the search does not consider liquidity
  or fees.

- [BTCPay Server #7354][] fixes a hot wallet key exposure issue introduced after
  [BTCPay Server #7329][] added granular wallet permissions. Users with
  wallet-signing permission, but not permission to view the wallet seed or modify
  store settings, could be exposed to derived hot wallet private keys during
  [PSBT][topic psbt] signing. The PR introduces a `HotwalletSafe` helper to
  centralize hot-wallet access, separates permission to sign from permission to
  view seed material, and updates signing flows to use the hot wallet
  server-side without returning private signing keys through HTTP form fields.

- [BIPs #2116][] publishes [BIP323][], which proposes expanding the number of
  bits available in `nVersion`'s nonce space for miners from 16 to 24,
  superseding [BIP320][]. It reserves bits 5 through 28 for header-only mining
  without relying on rolling `nTime` more often than once per second. See
  [Newsletter #395][news395 nversion] for previous discussion.

> 

Reviewer 2: The updates to BIP322 should be added. I’m gonna add a commit.

that could allow an attacker with sufficient proof-of-work to crash Bitcoin Core
nodes and describes a draft BIP proposal for sharing the UTXO set over the P2P
network. Also included are our regular sections announcing a new release
candidate and describing notable changes to popular Bitcoin infrastructure
software.

## News

- **Bitcoin Core script interpreter remote crash disclosure:**
  Niklas Gögge [posted][topic cve mailing list] to the Bitcoin-Dev mailing list
  disclosing [CVE-2024-52911][topic cve disclosure], a vulnerability affecting versions of Bitcoin Core
  after version 0.14.0 and before 29.0. After version 0.14.0 (released
  March 2017), validating a specially-crafted block could cause the node to access
  previously freed memory. During validation, data required for
  checking transaction inputs is cached. The bug occurred due to object lifetime
  ordering during parallel script validation, where cached precomputed
  transaction data could be freed before background script-check threads
  completed. For specially-crafted invalid blocks, it was possible for this data
  to be destroyed while it was still being accessed by background threads.

>   An attacker able to mine a valid block could specially craft it to crash a

Reviewer 3: I know you want to say that the attacker needs to have enough PoW (“able to mine”) but this makes it sound like the specially-crafted block is valid.

```suggestion
  An attacker with sufficient proof of work could, using the specially-crafted invalid block, crash a
```

  from the genesis block.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

- [Core Lightning 26.06rc1][] is a release candidate for the next major version
  of this popular LN node which includes new `graceful`, `sendamount`, and
  `xkeysend` RPCs, begins the `pay` deprecation cycle in favor of `xpay`, and
  adds BOLT12 payer-proof RPC support.

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #35209][] now constructs the `txsdata` vector before the
  `CCheckQueueControl` object, addressing the root cause of

>   [CVE-2024-52911][topic cves] (see the news section above). Since C++ destroys

Reviewer 4: ```suggestion
  [CVE-2024-52911][topic cve disclosure] (see the news section above). Since C++ destroys
```

---
title: 'Bitcoin Optech Newsletter #405'
permalink: /en/newsletters/2026/05/15/
name: 2026-05-15-newsletter
slug: 2026-05-15-newsletter
type: newsletter
layout: newsletter
lang: en
---
FIXME:bitschmidty

## News

- **Bitcoin Core CVE-2024-52911 Script Interpreter Remote Crash Disclosure:**
  Niklas Goegge [posted](topic cve mailing list) to the Bitcoin-Dev mailing list a link to an

>   [announcement](topic cve disclosure) of a security disclosure affecting versions of Bitcoin Core

Author: ```suggestion
  Niklas Gögge [posted][topic cve mailing list] to the Bitcoin-Dev mailing list a link to an
  [announcement][topic cve disclosure] of a security disclosure affecting versions of Bitcoin Core
```
